openapi-mock/server/mock_controller.js

```diff
@@ -16,9 +19,24 @@ module.exports = {
         for (var i = 0; i < config.apis.length; i++) {
             var api = config.apis[i];
             var openApi_doc = yaml.safeLoad(fs.readFileSync(api.specification_file, 'utf8'));
+            if (api.oauth)
+                ouath_endpoint = api.oauth;
+            if (api.metadata)
+                metadata_endpoint = api.metadata;
             createMetadataEndpoint(openApi_doc, api, app);
-            createEndpoints(openApi_doc, api);
             createConsole(openApi_doc, api, app);
+            if (api.hasOwnProperty("added_endpoints")) {

```

Comment:
why not moving all logic in the createEndpoints method? Or create another one to cover the file generation. Will simplify reading this main loop

Reply:
Moved all the logic to the createEndpoints method